grapher_admin/views.py

```diff
@@ -1434,3 +1547,359 @@ def register_by_invite(request: HttpRequest, code: str):
+        source_cursor.execute('select name, description from sources where id in (%s);' % ','.join([str(x) for x in source_ids]))
+        dataset_meta['sources'] = [{'name': item[0], 'description': item[1]} for item in source_cursor.fetchall()]
+
+    counter = 0
+    source_name_to_id = {}
+    for each in dataset_meta['sources']:
+        each['id'] = counter
+        source_name_to_id[each['name']] = counter
+        counter += 1
+
+    for each in vardata:
+        each['source_id'] = source_name_to_id[each['source_id']]
+
+    dataset_meta['variables'] = vardata
+
+    sql_query = 'SELECT `value`, `year`, data_values.`fk_var_id` as var_id, data_values.`fk_ent_id` as entity_id ' \
+                ' from data_values ' \
+                ' WHERE data_values.`fk_var_id` in (%s) AND ' \
+                'data_values.`fk_ent_id` in (%s) ORDER BY fk_ent_id, year, fk_var_id;'
+
+    metadata_filename = (unidecode(dataset.name) + '.json').replace('/', '_')
+    dataset_filename = (unidecode(dataset.name) + '.csv').replace('/', '_')
+
+    if old_dataset_name:
+        old_metadata_name = (unidecode(old_dataset_name) + '.json').replace('/', '_')
+        old_dataset_name = (unidecode(old_dataset_name) + '.csv').replace('/', '_')
+        old_metadata_file_path = os.path.join(current_dataset_folder, old_metadata_name)
+        old_dataset_file_path = os.path.join(current_dataset_folder, old_dataset_name)
+        old_metadata_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, old_metadata_name))
+        old_dataset_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, old_dataset_name))
+
+    metadata_file_path = os.path.join(current_dataset_folder, metadata_filename)
+    dataset_file_path = os.path.join(current_dataset_folder, dataset_filename)
+    metadata_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, metadata_filename))
+    dataset_file_path_escaped = shlex.quote(os.path.join(current_dataset_folder, dataset_filename))
+
+    delete_previous = False
+    if old_dataset_name is not None:
+        if os.path.isfile(old_dataset_file_path):
+            subprocess.check_output('git -C %s rm %s' %
+                                    (current_dataset_folder,
+                                     old_dataset_file_path_escaped),
+                                    shell=True)
+            subprocess.check_output('git -C %s rm %s' %
+                                    (current_dataset_folder,
+                                     old_metadata_file_path_escaped),
+                                    shell=True)
+            delete_previous = True
+            commit_message = 'Updating the dataset: %s'
+        else:
+            commit_message = 'Creating the dataset: %s'
+    else:
+        commit_message = 'Creating the dataset: %s'
+
+    # now saving the dataset to file
+
+    with open(dataset_file_path, 'w', newline='', encoding='utf8') as f:
```

Comment:
It would be wise to write the dataset to a /tmp/{hash} location first and then atomically `mv` into the repo after the write is completed, in the same command sequence as the commit. This would reduce the risk of data corruption due to interrupted write operations.